dias/tracker.py

```diff
@@ -0,0 +1,400 @@
+        ----------
+        analyzer : String
+            name of the analyzer to add
+        """
+        if not Analyzer.select().where(Analyzer.name == analyzer).exists():
+            Analyzer.create(name=analyzer)
+
+    def new_files(self, dias_task_name, filetypes, update=True):
+        """
+        Return a list of files unprocessed by dias_task_name, of its filetype of interest.
+
+        Optionally updates the File Table with the current state of `self.path`.
+
+        Parameters
+        ----------
+        dias_task_name : String
+            Name of task whose files you wish to track.
+        filetypes: List of String or String
+            list of filetypes of interest
+        update : Boolean
+            If true, updates the File Table with the current state of `self.path`.
+        """
+        if update:
+            self.update_file_table()
+
+        self.add_analyzer_ine(dias_task_name)
+
+        if isinstance(filetypes, str):
+            filetypes = [FileType.get(FileType.name == filetypes)]
+        elif isinstance(filetypes, (list, tuple)):
+            filetypes = [q for q in FileType.select().where(FileType.name << filetypes)]
+
+        dias_task_analyzer = Analyzer.get(Analyzer.name == dias_task_name)
+
+        # Select files processed by analyzer
+        files_processed = (
+            File.select()
+            .join(Processed)
+            .join(Analyzer)
+            .where(Processed.analyzer_id == dias_task_analyzer)
+        )
+
+        # SELECT files of analyzer.file_type_id and NOT Processed by Analyzer
+        # TODO Filter by date
+        files_unprocessed = (
+            File.select()
+            .where(File.id.not_in(files_processed))
+            .where(File.file_type_id << filetypes)
+            .where(File.exists)
+            .order_by(File.start_time.asc())
+        )
+
+        return [f.filepath for f in files_unprocessed]
+
+    def register_done(self, dias_task_name, list_of_files):
+        """
+        Register files in list_of_files as processed by dias_task_name.
+
+        Parameters
+        ----------
+        dias_task_name : String
+            name of the analyzer that you want to GET and SET file tracking information for.
+        list_of_files : List of Strings
+            List of filepaths to be registered as processed. They need to already exist in the database File.
+        """
+        self.add_analyzer_ine(dias_task_name)
+
+        files_processed = File.select().where(File.filepath.in_(list_of_files))
+
+        analyzer_id = Analyzer.get(Analyzer.name == dias_task_name)
+
+        record = namedtuple("record", "file_id analyzer_id")
+
+        records = [record(f, analyzer_id) for f in files_processed]
+
+        with db:
+            # SQLite limits bulk inserts to 100 at a time
+            for batch in chunked(records, 100):
+                Processed.insert_many(
+                    batch, fields=[Processed.file_id, Processed.analyzer_id]
```

Comment:
I'm pretty sure if you don't set `fields` it will insert all of the fields in the model, so you can remove that argument.